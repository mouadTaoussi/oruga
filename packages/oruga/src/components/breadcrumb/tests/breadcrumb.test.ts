import { describe, test, expect, afterEach } from "vitest";
import { enableAutoUnmount, mount } from "@vue/test-utils";
// import { nextTick } from "vue";
// import { setTimeout } from "timers/promises";

// import { OBreadcrumb } from "@/components/breadcrumb/Breadcrumb.vue";
import OBreadcrumb from "@/components/breadcrumb/Breadcrumb.vue";

describe("<Breadcrumb>", () => {
    enableAutoUnmount(afterEach);

    test("is called", () => {
        // const wrapper = mount(OBreadcrumb);
        const triggerHTML =
            '<breadcrumb-item tag="a" href="/">Home</breadcrumb-item>';
        const wrapper = mount(OBreadcrumb, {
            slots: {
                trigger: triggerHTML,
            },
        });
        expect(!!wrapper.vm).toBeTruthy();
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.attributes("data-oruga")).toBe("breadcrumb");
    });

    test("rendeer correctly", () => {
        // const wrapper = mount(OBreadcrumb);
        const triggerHTML =
            '<breadcrumb-item tag="a" href="/">Home</breadcrumb-item>';
        const wrapper = mount(OBreadcrumb, {
            slots: {
                trigger: triggerHTML,
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.classes("o-breadcrumb")).toBeTruthy();
    });
    test("render items", () => {
        // const triggerHTML = "<breadcrumb-item tag="a" href='/'>Home</breadcrumb-item>";
        const triggerHTML =
            '<breadcrumb-item tag="a" href="/">Home</breadcrumb-item>';

        // const wrapper = mount(OBreadcrumb, {
        //     slot: triggerHTML,
        // });
        const wrapper = mount(OBreadcrumb, {
            slots: {
                trigger: triggerHTML,
            },
        });
        const trigger = wrapper.find(".o-breadcrumb-item");
        expect(trigger.html()).toBe(triggerHTML);
        expect(trigger.exists()).toBeTruthy();
        // expect(trigger.text()).contain("Home");
    });
    test("is centered", () => {
        const triggerHTML =
            '<breadcrumb-item tag="a" href="/">Home</breadcrumb-item>';

        const wrapper = mount(OBreadcrumb, {
            slot: triggerHTML,
            props: { align: "centered" },
        });

        expect(wrapper.classes("o-breadcrumb__centered")).toBeTruthy();
    });
    test("is large ", () => {
        const triggerHTML =
            '<breadcrumb-item tag="a" href="/">Home</breadcrumb-item>';

        const wrapper = mount(OBreadcrumb, {
            slot: triggerHTML,
            props: { size: "large" },
        });

        expect(wrapper.classes("o-breadcrumb__large")).toBeTruthy();
    });
});
