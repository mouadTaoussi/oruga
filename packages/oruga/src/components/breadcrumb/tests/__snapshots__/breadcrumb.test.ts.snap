// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Breadcrumb> > rendeer correctly 1`] = `
"<section class="o-breadcrumb" data-oruga="breadcrumb">
  <!-- BreadcrumbItems -->
</section>"
`;
